maven-model-transform/src/main/java/org/apache/maven/model/transform/pull/BufferingParser.java

```diff
@@ -430,7 +430,7 @@ public int nextToken() throws XmlPullParserException, IOException
                 throw new XmlPullParserException( "already reached end of XML input", this, null );
             }
             int currentEvent = xmlPullParser.nextToken();
-            if ( accept() )
+            if ( disabled || accept() )
```

Comment:
This looks like to conflicting states, which makes me wonder if we can always return currentEvent.
If current expression is required, we need to think of better naming.

Reply:
Maybe rename `disabled` to `bypass` ?
Or `accept()` to `canYieldCurrentEvent()` ?
Or both...

Reply:
I like `bypass` - and I'd like it even more if the variable name would be in the same terminology as the class Javadoc. So maybe we should rephrase the Javadoc a bit and say something like "This filter will bypass all following filters and write directly to the output."